### PR TITLE
Bluetooth: controller: re-add missing hci_vendor.c reference

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Add hci/nordic/hci_vendor.c back to the list when building for nRF platforms, this was inadvertently removed in

7bd2804c24 Bluetooth: Controller: Build vendor LLL as library

and is currently causing a:

<wrn> bt_id: No static addresses stored in controller

and random address on nRF devices.

### CODE REVIEW

File: subsys/bluetooth/controller/CMakeLists.txt
Comment (cvinayak) [inline subsys/bluetooth/controller/CMakeLists.txt:194]: Thank you for discovering.

1. Add it here: https://github.com/cvinayak/zephyr/blob/2de71fa1d7eb3e5b29b52dd0be9d1f6eb8b52572/subsys/bluetooth/controller/ll_sw/nordic/CMakeLists.txt#L87 and respectively for `openisa` too.

as:
```c
zephyr_library_sources(
  hci/hci_vendor.c
  )
```

2. Move the vendor specific {nordic, openisa} header and source files too into `ll_sw/{nordic,openisa}/hci/` 

(maybe easy said by me, but give a try else I can do it).

**Review by cvinayak — DISMISSED** [review]

File: subsys/bluetooth/controller/CMakeLists.txt
Comment (fabiobaltieri) [reply subsys/bluetooth/controller/CMakeLists.txt]: okay, had to adjust a path as well, did I get this right?

**Comment by fabiobaltieri** [comment]
@cvinayak applied the comment but also noticed that the previous twister run failed on `nrf52_bsim` with

```
/usr/bin/ld.bfd: zephyr/subsys/bluetooth/controller/libsubsys__bluetooth__controller.a(hci_vendor.c.obj): in function `hci_vendor_read_static_addr':
/__w/zephyr/zephyr/subsys/bluetooth/controller/hci/nordic/hci_vendor.c:19: multiple definition of `hci_vendor_read_static_addr'; app/libapp.a(hci_vendor_stub.c.obj):/__w/zephyr/zephyr/tests/bluetooth/df/common/src/hci_vendor_stub.c:17: first defined here
```

Guess that file should be skipped when building for bsim, how should it be handled? unsure

**Comment by fabiobaltieri** [comment]
Tentatively put the file behind a `if (NOT CONFIG_SOC_SERIES_BSIM_NRFXX)`

**Comment by cvinayak** [comment]
> Tentatively put the file behind a `if (NOT CONFIG_SOC_SERIES_BSIM_NRFXX)`

Sorry my oversight, previously it was like this:
```c
zephyr_library_sources_ifdef(
  CONFIG_SOC_FAMILY_NRF
  hci/nordic/hci_vendor.c
  )
```

Maybe the below is better (and works as intended for both SoC and BSIM?):
```c
zephyr_library_sources_ifdef(
  CONFIG_SOC_FAMILY_NRF
  hci/hci_vendor.c
  )
```

@aescolar Need your review here as to how it should be these days...

**Comment by cvinayak** [comment]
I will push the change and see...

**Comment by fabiobaltieri** [comment]
Cool, CI is happy

**Comment by aescolar** [comment]
> @cvinayak applied the comment but also noticed that the previous twister run failed on nrf52_bsim with

@fabiobaltieri Do you recall or can you locate which test was failing?

**Comment by aescolar** [comment]
> @aescolar Need your review here as to how it should be these days...

By now I guess we can just get this is as is, as this issue of not building this file for simulation was always there.
I'd like to fix it, but given what Fabio said above, it may be better to follow it up on another PR

**Comment by fabiobaltieri** [comment]
> @fabiobaltieri Do you recall or can you locate which test was failing?

None of them were failing, I noticed the addressed changing randomly while working on an out-of-tree project and reproduced with the beacon one but could have used any of those.

> By now I guess we can just get this is as is, as this issue of not building this file for simulation was always there.

Yeah this is annoying to anyone working on main, may be a good idea to get it in sooner rather than later.

**Comment by aescolar** [comment]
@fabiobaltieri @cvinayak  I think this is the proper fix on top of this PR so this file is also built for simulation:
https://github.com/aescolar/zephyr/commit/93ec925fc3d4bb4dc958187dd765c5504067f8ae

**Review by aescolar — DISMISSED** [review]
I'm ok with this being merged as is.
You can also take https://github.com/aescolar/zephyr/commit/93ec925fc3d4bb4dc958187dd765c5504067f8ae into this PR if you want,
or I can submit a follow up with that commit right after this one is merged.
You decide

**Review by aescolar — APPROVED** [review]

**Comment by fabiobaltieri** [comment]
> You can also take [aescolar@93ec925](https://github.com/aescolar/zephyr/commit/93ec925fc3d4bb4dc958187dd765c5504067f8ae) into this PR if you want,

Yup, pulled that in this PR.

**Review by cvinayak — APPROVED** [review]

**Comment by cvinayak** [comment]
Minor nitpick on the title: `subsys/bt/ctlr/nordic: Build hci_vendor.c also for simulation`  to be `Bluetooth: Controller: Build hci_vendor.c also for simulation`

**Comment by fabiobaltieri** [comment]
> `Bluetooth: Controller: Build hci_vendor.c also for simulation`

Fixed

**Pull request merged**
